woocommerce/woocommerce-ios
# Variations: Add UI for rename attributes flow

Related: #3212 

## Description

This PR adds the UI for renaming variation attributes:

* Adds a Rename button in the options menu.
* Builds and presents the Rename Attribute screen.

This only adds the UI; the changes to actually perform the update (to rename the attribute) will come in another PR.

## Changes

* Updates `AddAttributeOptionsViewController` to add a Rename button to the `...` menu action sheet. Tapping the Rename button navigates to the Rename Attribute screen.
* New `RenameAttributesViewController`, `RenameAttributesViewController.xib`, and `RenameAttributesViewModel` for the Rename Attribute screen:
   * Rename Attribute screen displays a single text field for the attribute name, prefilled with the current attribute name.
   * Done button is disabled if there is no text in the text field (to ensure the attribute has a name).
   * For now, tapping Done just navigates back. (This will be updated to perform the update in another PR.)
   * Checks for unsaved changes and presents "discard changes" action sheet when navigating back.

Rename button|Rename Attribute screen
-|-
![Simulator Screen Shot - iPhone 12 - 2021-03-01 at 11 31 56](https://user-images.githubusercontent.com/8658164/109491942-2335d600-7a82-11eb-8edc-bf128f53cbd9.png)|![Simulator Screen Shot - iPhone 12 - 2021-03-01 at 11 32 00](https://user-images.githubusercontent.com/8658164/109491946-24670300-7a82-11eb-8dd0-926cbeffed8d.png)


<img src="https://user-images.githubusercontent.com/8658164/109491925-1d3ff500-7a82-11eb-9b31-5b83497a44cb.gif" width="300px">


## Testing

1. Go to the Products tab.
2. Select a variable product.
3. Select Variations to open the variations list.
4. Tap the `...` menu and select Edit Attributes.
5. Select an attribute to open the Attribute Options screen.
6. Tap the `...` menu and select Rename.
7. Confirm the Rename Attribute screen opens as expected, with the text field prefilled with the current name of the attribute you selected.
8. Confirm you can enter a new attribute name in the text field, and the Done button is disabled if you remove all the text there.

## Submitter Checklist

Update release notes:
